soroban-env-host/src/test/poseidon/poseidon2_instance_bls12.rs

```diff
@@ -0,0 +1,1532 @@
+use crate::crypto::poseidon::Poseidon2Params;
+use crate::testutils::crypto::from_hex;
+use ark_bls12_381::Fr as Scalar;
+use std::sync::Arc;
+
+lazy_static::lazy_static! {
+    pub static ref MAT_DIAG2_M_1: Vec<Scalar> = vec![
+    from_hex("0x0000000000000000000000000000000000000000000000000000000000000001"),
+    from_hex("0x0000000000000000000000000000000000000000000000000000000000000002"),
+    ];
+
+    // the full matrix is provided (in the original test file) but not used
+    pub static ref MAT_INTERNAL2: Vec<Vec<Scalar>> = vec![
+    vec![from_hex("0x0000000000000000000000000000000000000000000000000000000000000002"),
+    from_hex("0x0000000000000000000000000000000000000000000000000000000000000001"),
+    ],
+    vec![from_hex("0x0000000000000000000000000000000000000000000000000000000000000001"),
+    from_hex("0x0000000000000000000000000000000000000000000000000000000000000003"),
+    ],
+    ];
+
+    pub static ref RC2: Vec<Vec<Scalar>> = vec![
```

Comment:
Do we have any references to the source of these test vectors? It would be nice to add the references where applicable.